### ISSUE TITLE
Error with multiple merge bases

### ISSUE THREAD

**Issue by raymondhoagland** [danger/danger#965]
# Report

## What did you do?
Run Danger on a PR from a Jenkins build.

## What did you expect to happen?
Danger runs without crashing.

## What happened instead?
In _some_ scenarios, Danger crashes with an error remarkably similar to the error in #904.

## Your Environment
* Running in Jenkins
* Running on v5.5.5
  ```ruby
  # please paste here, with 2-space identation, thank you!
  xcov.report(
     scheme: ENV["FL_XCOV_SCHEME"],
     workspace: ENV["FL_XCOV_WORKSPACE"],
     skip_slack: ENV["FL_XCOV_SKIP_SLACK"],
     derived_data_path: ENV["FL_XCOV_DERIVED_DATA"],
     only_project_targets: true
  )
  xcode_summary.ignored_files="Pods/**"
  xcode_summary.report(ENV["FL_XCS_REPORT"])
  ``` 
Not exactly the right Dangerfile but I don't think it's important.  If it is, I'll sanitize it and update.

## Food For Thought
My best guess for how this ended up happening is related to [this line](https://github.com/danger/danger/blob/0d70bc4560976f30ca30dd3e7d2582d944bc4a83/lib/danger/scm_source/git_repo.rb#L168), specifically the fact that the call is for `merge-base --all` which has the chance of providing multiple merge bases.  

In most cases there should be a single merge base returned because the incremental fetch starts out very small.  However if the incremental fetch size increases because a merge base was not found, it may end up becoming large enough that multiple merge bases are returned at once.  The result (multiple lines of text) isn't sanitized so the error ends up getting displayed.

This normally ends up sorting itself out as a merge base eventually appears closer to the tip of the branch and only one result is found again, making it harder to debug.  I think this happens to me more often because I am using shallow cloning - the merge base isn't found right away and a larger fetch is often required.

Is there a reason to include the `--all` flag?  If I'm right removing that would easily solve this and provide a much better explanation for #904 than not using a puts statement.

**Comment by tbrand** [comment]
Hi @raymondhoagland

I've created the patch at https://github.com/danger/danger/pull/967
BTW, I've fixed it by chnaging a way to fetch the repositories like this
```
git init
git remote add origin <some origin>
git fetch origin <base_branch> --depth 1
git fetch origin +refs/pull/<pr_number>/merge
git reset --hard FETCH_HEAD
```

It's inspired by [Travis-CI](https://travis-ci.org/danger/danger/jobs/351197498#L415)

Regards.

**Comment by raymondhoagland** [comment]
Fixed with #967
